src/util/artifacts.js

```diff
@@ -14,12 +16,36 @@ export async function prepareArtifactsDir(
     const stats = await fs.stat(artifactsDir);
     if (!stats.isDirectory()) {
       throw new UsageError(
-        `--artifacts-dir=${artifactsDir} (this value is not a directory)`);
+        `"${artifactsDir}" exists and it is not a directory.`);
+    }
+    // If the artifactsDir already exists, check that we have the write permissions on it.
+    try {
+      await fs.access(artifactsDir, fs.W_OK);
+    } catch (accessErr) {
+      if (isErrorWithCode('EACCES', accessErr)) {
+        throw new UsageError(
+          `"${artifactsDir}" exists but the user lacks permissions on it.`);
+      } else {
+        throw accessErr;
+      }
     }
   } catch (error) {
-    if (isErrorWithCode('ENOENT', error)) {
-      log.debug(`Creating artifacts directory: ${artifactsDir}`);
-      await fs.mkdir(artifactsDir);
+    if (isErrorWithCode('EACCES', error)) {
+      // Handle errors when the artifactsDir cannot be accessed.
+      throw new UsageError(
+        `Cannot access "${artifactsDir}", user lacks permissions.`);
```

Comment:
The prefix will help provide context. I also think the reason could be made more specific:

    `Cannot access --artifacts-dir="${artifactsDir}" because the user lacks permissions.`